.github/workflows/compile.yml

```diff
@@ -0,0 +1,77 @@
+name: Build policies
+on:
+  - pull_request
+  - push
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        name:
+          - Fedora 33 GCC
+          - Fedora 33 GCC strict
+          - Fedora 33 Clang
+          - Fedora 33 Clang strict
+          - Fedora 33 Clang -Weverything
+
+        include:
+          - name: Fedora 33 GCC
+            image: fedora:33
```

Comment:
I haven't tried yet but can we use Fedora 34 here?
There is image fedora:34 but I haven't checked if there will be any other issues.

In general, I think, it's better to test 389-ds 'master' on the distro for which we build RPMs.

Reply:
> I haven't tried yet but can we use Fedora 34 here?
There is image fedora:34 but I haven't checked if there will be any other issues.

We can, but it's not branched yet, it's still rawhide. We expect some unrelated failures due to instability of rawhide. I'd prefer to add F34 once it's branched.

> In general, I think, it's better to test 389-ds 'master' on the distro for which we build RPMs - master==fedora-rawhide.

The point of this test is to use stable tools (gcc, clang) to test unstable code (master). If tools are from an unstable version, we can get unpredictable results. I can add another variant to the matrix, but I'm not sure how useful it would be.

Reply:
Okay, sounds good and I think you're right.

The only issues I'm worried about are the Fedora changes related issues (like crypto policies may be different on F33 and rawhide).
But it's really rare and we can catch it separately.